internal/compatibility/proto/request.go

```diff
@@ -81,6 +81,16 @@ func DescribeWorkflowExecutionRequest(t *shared.DescribeWorkflowExecutionRequest
 	}
 }
 
+func DiagnoseWorkflowExecutionRequest(t *shared.DiagnoseWorkflowExecutionRequest) *apiv1.DiagnoseWorkflowExecutionRequest {
```

Comment:
I am not super happy with this API's response. It just returns wfid/runid corresponding to the diagnostic workflow which means user has to make subsequent API calls to get diagnosis results. I think this was designed to be used by CLI but we have to expose it here in SDKs too unfortunately.